tracetypes/org.eclipse.tracecompass.incubator.opentracing.core/src/org/eclipse/tracecompass/incubator/opentracing/core/analysis/callstack/SpanCallGraphAnalysis.java

```diff
@@ -0,0 +1,178 @@
+        Collection<ICallStackElement> rootElements = callstackSerie.getRootElements();
+        for (ICallStackElement element : rootElements) {
+            if (monitor.isCanceled()) {
+                return false;
+            }
+            iterateOverElement(element, model, callgraph, start, end, monitor);
+        }
+        return true;
+    }
+
+    private void iterateOverElement(ICallStackElement element, IHostModel model, CallGraph callgraph, long start, long end, IProgressMonitor monitor) {
+        // Iterator over the children of the element until we reach the leaves
+        if (element.isLeaf()) {
+            iterateOverLeafElement(element, model, callgraph, start, end, monitor);
+            return;
+        }
+        for (ICallStackElement child : element.getChildrenElements()) {
+            iterateOverElement(child, model, callgraph, start, end, monitor);
+        }
+    }
+
+    private void iterateOverLeafElement(ICallStackElement element, IHostModel model, CallGraph callgraph, long start, long end, IProgressMonitor monitor) {
+        if (!(element instanceof InstrumentedCallStackElement)) {
+            throw new IllegalStateException("Call Graph Analysis: The element does not have the right type"); //$NON-NLS-1$
+        }
+        InstrumentedCallStackElement insElement = (InstrumentedCallStackElement) element;
+
+        CallStack callStack = insElement.getCallStack();
+
+        // If there is no children for this callstack, just return
+        if (callStack.getMaxDepth() == 0) {
+            return;
+        }
+        fHasKernelStatuses |= callStack.hasKernelStatuses();
+        // Start with the first function
+        AbstractCalledFunction nextFunction = (AbstractCalledFunction) callStack.getNextFunction(callStack.getStartTime(), 1, null, model, start, end);
+        while (nextFunction != null) {
+            AggregatedCalledFunction aggregatedChild = createSpanCallSite(CallStackSymbolFactory.createSymbol(getFuncName(nextFunction.getSymbol()), element, nextFunction.getStart()));
+            iterateOverCallstack(element, callStack, nextFunction, 2, aggregatedChild, model, start, end, monitor);
+            aggregatedChild.addFunctionCall(nextFunction);
+            Iterable<ProcessStatusInterval> kernelStatuses = callStack.getKernelStatuses(nextFunction, Collections.emptyList());
+            for (ProcessStatusInterval status : kernelStatuses) {
+                aggregatedChild.addKernelStatus(status);
+            }
+            callgraph.addAggregatedCallSite(element, aggregatedChild);
+            nextFunction = (AbstractCalledFunction) callStack.getNextFunction(nextFunction.getEnd(), 1, null, model, start, end);
+        }
+    }
+
+    private void iterateOverCallstack(ICallStackElement element, CallStack callstack, ICalledFunction function, int nextLevel, AggregatedCalledFunction aggregatedCall, IHostModel model, long start, long end, IProgressMonitor monitor) {
```

Comment:
Formatting should be corrected in this file (this can be done simply with ctrl+shift+f in eclipse)